packages/components/data-resource/reducer.tsx

```diff
@@ -0,0 +1,33 @@
+import { createFormAction, initialStateCreateForm } from './constants';
+import { CreateFormAction, CreateFormDataState } from './types';
+
+export const createFormReducer = (
+  state: CreateFormDataState,
+  action: CreateFormAction
+) => {
+  const { value } = action;
+  switch (action.type) {
+    case createFormAction.SET_NAME:
+      return Object.assign({}, state, { name: value });
+    case createFormAction.SET_PROVIDER:
+      return Object.assign({}, state, { provider: value });
+    case createFormAction.SET_SECRET_NAME:
+      return Object.assign({}, state, { secretName: value });
+    case createFormAction.SET_SECRET_KEY:
+      return Object.assign({}, state, { secretKey: value });
+    case createFormAction.SET_ACCESS_KEY:
+      return Object.assign({}, state, { accessKey: value });
+    case createFormAction.SET_REGION:
+      return Object.assign({}, state, { region: value });
+    case createFormAction.SET_TARGET:
+      return Object.assign({}, state, { target: value });
+    case createFormAction.SET_END_POINT:
+      return Object.assign({}, state, { endpoint: value });
+    case createFormAction.SET_PVC:
+      return Object.assign({}, state, { pvc: value });
+    case createFormAction.SET_PVC_FOLDER_NAME:
+      return Object.assign({}, state, { pvcFolderName: value });
+    default:
+      return initialStateCreateForm;
+  }
+};
```

Comment:
remove `types.ts` and `constant.ts` and add them here... else there will be circular dependency...